packages/yarnpkg-core/sources/tgzUtils.ts

```diff
@@ -39,11 +40,11 @@ export interface ExtractBufferOptions {
 
 let workerPool: WorkerPool<ConvertToZipPayload, PortablePath> | null;
 
-export async function convertToZip(tgz: Buffer, opts: ExtractBufferOptions) {
+export async function convertToZip(tgz: Buffer, limit: Limit, opts: ExtractBufferOptions) {
```

Comment:
Can you move `limit` inside the `opts` object? And perhaps renaming it `poolSize`